lib/graphql/enum_type.rb

```diff
@@ -136,7 +138,7 @@ def initialize(name:, description:, deprecation_reason:, value:)
         @name = name
         @description = description
         @deprecation_reason = deprecation_reason
-        @value = value
+        @value = value || name
```

Comment:
So this is the fix for the enum dump + load bug, actually. Here's the bug:

In loader.rb, when an enum is being loaded, it receives a hash that looks like this:
```ruby
{"kind"=>"ENUM",
 "name"=>"Choice",
 "description"=>nil,
 "fields"=>nil,
 "inputFields"=>nil,
 "interfaces"=>nil,
 "enumValues"=>
  [{"name"=>"FOO", "description"=>nil, "isDeprecated"=>false, "deprecationReason"=>nil}, {"name"=>"BAR", "description"=>nil, "isDeprecated"=>false, "deprecationReason"=>nil}],
 "possibleTypes"=>nil}
```

It then instantiates `EnumType::EnumValue` and passes `value: enum['value']` to the constructor. Well, the thing is, application values for these enum's is not included in the introspection query. To my knowledge, even having application values is just an implementation detail that is not defined in the GraphQL Spec. 

The result is that the constructor gets `value: nil` passed in, and that causes all kinds of problems (eg, `@values_by_value` ends up looking like `{ nil => EnumValue<...> }`).

I modified it here, but now that I'm looking at it, this fix should probably be in the loader instead. But does that make sense?